### ISSUE TITLE
Api Error are black boxes. Which is really bad when debugging.

### ISSUE THREAD

**Issue by vmeyet** [paidapi/paid-ruby#7]
Whatever the API error the response is "Invalid request. Please check the URL and parameters." if the http code is not_found (404) or bad_request (400).

The paid api does return interesting information, why hiding it?

ex:

```
400 Bad Request: {"error":{"type":"invalid_request_error","message":"Customer does not have any invoiceable transactions."}}
```

Is way more clearer than

``` ruby
>>> customer.generate_invoice
# > Paid::APIError: Invalid request. Please check the URL and parameters.
# > from /Users/vivien/.rbenv/versions/2.0.0-rc2/lib/ruby/gems/2.0.0/gems/paid-1.0.5/lib/paid/api_method.rb:36:in `rescue in execute'
```

(or for empty descriptions when creating a transaction, etc)

**Comment by joncalhoun** [comment]
You are right - we should just put the server response in the error message.

In the meantime, the json from the server *is* available. Simply rescue the error, then call `error.json`. Eg:

```ruby
Paid.api_key = "invalid-api-key"
begin
  Paid::Customer.all
rescue Exception => e
  puts e.json # {:error=>{:type=>"invalid_request_error", :message=>"Invalid API token"}}
  puts e.json[:error][:message]  # Invalid API token
end
```

**Comment by vmeyet** [comment]
Thanks for the rapid answer.

Indeed I missed that, we can access the api response when catching the exception. Still a pain when using `pry` or when logging the error (in a file/sentry/whatever) though.

**Comment by joncalhoun** [comment]
Yeah, we need to just make the message match the server response when there is one, and it will be there with a bump to v2.0.0 (we are changing some underlying base code, so opting for a major bump since a few subtle changes don't work exactly the same between versions).

**Comment by ryanwjackson** [comment]
This was resolved with #13 (v1.2.0).  Please open a new issue if you expect different behavior.
